apps/Learning-Buddy/screens/SignUpScreen.jsx

```diff
@@ -16,45 +16,85 @@ export const SignUpScreen = ({ navigation }) => {
   const [phone, setPhone] = useState();
   const [password, setPassword] = useState();
   const [password2, setPassword2] = useState();
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [password2Error, setPassword2Error] = useState(false);
   const styles = useContext(StyleSheetContext);
   const auth = useContext(AuthContext);
   const { signUpWithEmail } = auth;
 
   const formVerify = () => {
-    if (name && email && password && password2) {
-      if (password === password2) {
-        if (isValidEmail(email)) {
-          if (checked) {
-            const returnVal = {
-              name,
-              email,
-              password,
-              ...(phone && { phone })
-            };
+    if (!name && !email && !password && !password2) {
+      setNameError(true);
+      setEmailError(true);
+      setPasswordError(true);
+      setPassword2Error(true);
+    }
 
-            return returnVal;
-          } else {
-            Alert.alert('Please accept our Terms of Service');
-          }
-        } else {
-          Alert.alert('Invalid email address');
-        }
-      } else {
-        Alert.alert("Passwords don't match");
-      }
-    } else {
-      Alert.alert('Missing a required field');
+    if (!name) {
+      setNameError(true);
+      return false;
+    }
+
+    if (!email || !isValidEmail(email)) {
+      setEmailError(true);
+      return false;
+    }
+
+    if (!password) {
+      setPasswordError(true);
+      return false;
+    }
+    if (!password2) {
+      setPassword2Error(true);
+      return false;
+    }
+
+    if (password !== password2) {
+      setPasswordError(true);
+      setPassword2Error(true);
+      return false;
+    }
+
+    if (!checked) {
+      Alert.alert('Please accept our Terms of Service');
+      return false;
     }
+
+    const returnVal = {
+      name,
+      email,
+      password,
+      ...(phone && { phone })
+    };
+    return returnVal;
   };
 
   const handleSubmitSignUp = async () => {
-    if (formVerify()) {
-      const status = await signUpWithEmail(email, password, name, phone);
-      if (status === 'SignedUp') {
-        Alert.alert('Success! You Are Signed Up!');
-      } else {
-        Alert.alert(status);
+    setNameError(false);
+    setEmailError(false);
+    setPasswordError(false);
+    setPassword2Error(false);
```

Comment:
These can be placed inside of the form validation at the top of the function.

Reply:
I moved it